### PR TITLE
DOC-10369 - Add Legacy Durability documentation

### PR DESCRIPTION
**Pull request by maria-robobug**
https://issues.couchbase.com/browse/DOC-10369

Added simple example showing `persist_to` and `replicate_to` legacy durability options and documentation. Similar to what we do in Java [here](https://docs.couchbase.com/java-sdk/current/howtos/kv-operations.html#durability).

### CODE REVIEW

File: modules/howtos/examples/kv_operations.rb [hunk 1]
```diff
@@ -92,6 +92,13 @@
   puts "Example requires a multi-node environment"
 end
 
+begin
+  # tag::durability-observed[]
+  collection.upsert("my-document", {"doc" => true}, 
+                  Options::Upsert(persist_to: :none, replicate_to: :two))
+  # end::durability-observed[]
+rescue Error::DurabilityImpossible
```
Comment (maria-robobug) [inline modules/howtos/examples/kv_operations.rb:100]: Seems we will have Legacy Durability in Ruby SDK 3.4.

@avsej I'm just prepping this example before the release, let me know if anything looks incorrect. I simply tried to match the example we have in [Java](https://github.com/couchbase/docs-sdk-java/blob/d3c6d9cd0b3b9e8aa5d4a785da7f2cd73c2fc5c2/modules/howtos/examples/KvOperations.java#L181-L182).

File: modules/howtos/pages/kv-operations.adoc [hunk 2]
```diff
@@ -103,7 +112,7 @@ and the default level of safety provided by Couchbase will be reasonable for the
 .Sub-Document Operations
 ====
 All of these operations involve fetching the complete document from the Cluster.
-Where the number of operations or other circumstances make bandwidth a significant issue, the SDK can work on just a specific _path_ of the document with xref:subdocument-operations.adoc[Sub-Docunent Operations].
+Where the number of operations or other circumstances make bandwidth a significant issue, the SDK can work on just a specific _path_ of the document with xref:subdocument-operations.adoc[Sub-Document Operations].
 ====
 
 == Retrieving full documents
```
Comment (RichardSmedley) [inline modules/howtos/pages/kv-operations.adoc:118]: There's mention of 7.0β further down - can we get this out in the same PR?

File: modules/howtos/pages/kv-operations.adoc [hunk 3]
```diff
@@ -65,34 +65,43 @@ xref:7.1@server:learn:buckets-memory-and-storage/expiration.adoc#expiration-buck
...
+* `:majority_and_persist_to_active` - Majority level, plus persisted to disk on the active node.
+* `:persist_to_majority` - Majority level, plus persisted to disk on the majority of configured replicas.
 
 The options are in increasing levels of safety.
 Note that nothing comes for free - for a given node, waiting for writes to storage is considerably slower than waiting for it to be available in-memory.
 These trade offs, as well as which settings may be tuned, are discussed in the xref:concept-docs:durability-replication-failure-considerations.adoc#durable-writes[durability page].
 
-The following example demonstrates using the newer durability features available in Couchbase server 6.5 onwards.
-
+If a version of Couchbase Server lower than 6.5 is being used then the application can fall-back to xref:concept-docs:durability-replication-failure-considerations.adoc#older-server-versions['client verified' durability].
```
Comment (RichardSmedley) [inline modules/howtos/pages/kv-operations.adoc]: I think we had "lower" proscribed in the earlier style guide that we inherited from marketing.
Will add it to current style discussion.  :+1:

**Review by RichardSmedley — APPROVED** [review]
Couple of comments, but LGTM.
Grazie @maria-robobug   :+1:

File: modules/howtos/pages/kv-operations.adoc (hunk 2 above)
Comment (maria-robobug) [reply modules/howtos/pages/kv-operations.adoc:118]: Ooh, hadn't spotted, absolutely! 👍

**Review by RichardSmedley — APPROVED** [review]
LGTM  :+1:

**Comment by maria-robobug** [comment]
Going to merge this so we can see changes in Staging, but if @avsej has any comments I can always update.

**Comment by avsej** [comment]
Staging is good, although the change hasn't been released in the SDK yet (only on `main` branch)

**Comment by maria-robobug** [comment]
> Staging is good, although the change hasn't been released in the SDK yet (only on main branch)

Yep, indeed, this will only be seen in staging 👍 .

**Pull request merged**